File: week1_programming_challenges/2_maximum_pairwise_product/max_pairwise_product.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
});

process.stdin.setEncoding('utf8');
rl.once('line', () => {
    rl.on('line', readLine);
});

function readLine (line) {
    const arr = line.toString().split(' ').map(Number);

    console.log(max(arr));
    process.exit();
}

function max(arr) {
    arr.sort(function(a, b){return a - b});
    var arr_size = arr.length;
    return arr[arr_size-1]*arr[arr_size-2];
}

module.exports = max;

File: week1_programming_challenges/2_maximum_pairwise_product/max_pairwise_product.py
def max_pairwise_product(numbers):
    assert len(numbers) >= 2
    assert all(0 <= x <= 2 * 10 ** 5 for x in numbers)
    numbers.sort()
    n = len(numbers)
    max_product = numbers[n - 1] * numbers[n - 2]
    return max_product


if __name__ == '__main__':
    n = int(input())
    input_numbers = list(map(int, input().split()))
    assert len(input_numbers) == n
    print(max_pairwise_product(input_numbers))
